### PR TITLE
Added AWS EKS integration

### PR DESCRIPTION
**Pull request by ltflb-bgdi**
## Commit message
Add EKS support to AWS integration

EKS is the managed kubernetes service of AWS. Since there is no EKS integration available, users are forced to implement
[custom solutions](https://github.com/elastic/integrations/issues/5799) to ingest EKS logs. This is a hard to maintain and error prone task.  

There are 5 types of EKS logs, where 4 of them are standard kubernetes logs.

The approach of this integration is to add two datastreams to the AWS integration.
- eks_logs: Used for message parsing and rerouting. The kubernetes logs are forwarded to the corresponding kubernetes datastreams.
- eks_authenticator_logs: Used to store authenticator logs, which are EKS specific.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist
- [x] audit log support
- [x] API server log support
- [x] Authenticator log support
- [x] Controller manager log support
- [x] Event (logs) support 
- [x] Scheduler log support
- [x] Add EKS support to awsfirehose data_stream
- [ ] Update/Review documenation
- [ ] Update/Review manifest
- [x] Add kubernetes API server log data_stream
- [x] Add controller manager log data_stream
- [x] Add kubernetes scheduler log data_stream
- [x] Add kubernetes event log data_stream

## How to test this PR locally

```
cd packages/aws
elastic-package stack up -d
elastic-package test pipeline
```

## Related issues

- Relates #5799

### CODE REVIEW

**Comment by ltflb-bgdi** [comment]
@andrewkroh I have some questions on kubernetes events datastream:
- Is this a metrics datastream by intention? It seems to me more as a log datastream. Is this due to the fact that metricbeat support polling?
- Is it intended that fields are under kubernetes.events, even those who could go under generic fields, like message, kubernetes.pod.name etc.?
- It the current implementation compatible with logs shipped from other sources like https://github.com/resmoio/kubernetes-event-exporter

**Comment by andrewkroh** [comment]
@elastic/obs-cloudnative-monitoring, as owners of the kubernetes integration, can I redirect those ^ [questions](https://github.com/elastic/integrations/pull/10521#issuecomment-2298891235) to you please.

**Review by dej611 — APPROVED** [review]
Visualization code owner files 👍

File: packages/aws/data_stream/vpcflow/manifest.yml [hunk 1]
```diff
@@ -70,12 +70,8 @@ streams:
...
-          We recommend to keep the default value 5 and use the [preset](https://www.elastic.co/guide/en/fleet/current/es-output-settings.html#es-output-settings-performance-tuning-settings)
-          option to tune your Elastic Agent performance. You can optimize for throughput, scale, latency, or you can
-          choose a balanced (the default) set of performance specifications.
+          The maximum number of SQS messages that can be inflight at any time. Defaults to 5. When processing large  amount of large size S3 objects and each object has large amount of events, if this parameter sets too high, it can cause the input to process too many messages concurrently, overload the agent and cause ingest failure. We recommend to keep the default value 5 and use the [preset](https://www.elastic.co/guide/en/fleet/current/es-output-settings.html#es-output-settings-performance-tuning-settings) option to tune your Elastic Agent performance. You can optimize for throughput, scale, latency, or you can choose a balanced (the default) set of performance specifications.
```
Comment (andrewkroh) [inline packages/aws/data_stream/vpcflow/manifest.yml:73]: There was no content change here so I recommend reverting this, and the similar mods. That will reduce the number of reviewable files (and teams needed to sign-off).

Comment (ltflb-bgdi) [reply packages/aws/data_stream/vpcflow/manifest.yml:73]: Done. Please note that this was done by the `elastic-package format` command. Formatters should fix and not mess up formats IMO. Please consider fixing it.

**Comment by ltflb-bgdi** [comment]
> @elastic/obs-cloudnative-monitoring, as owners of the kubernetes integration, can I redirect those ^ [questions](https://github.com/elastic/integrations/pull/10521#issuecomment-2298891235) to you please.

@elastic/obs-cloudnative-monitoring Anybody?

Please note that I have added a new `event_logs data_stream` in the meantime, collected with the recently launched `elastic-otel-collector`. The main reason is to have the k8s cluster name to properly set data_stream namespaces and the correct data_stream type (logs instead of metrics). In addition it contains some ECS parsing.

**Comment by gizas** [comment]
> @elastic/obs-cloudnative-monitoring Anybody?

Sorry for the delay here, seems that we missed the ping.  To your questions:

It seems that the history of the events creation goes back to https://github.com/elastic/beats/pull/4315. I dont have all the justification but seems that because the events are coming through polling maybe this seemed to be the same with other metricsets where we are polling some kind of endpoint. My personal view (also in the new otel collector), is that events match to go under logs. 

For the second question, yes is intended to have them all under kubernetes.events, to have the same structure as we have with other metricsets of the k8s integration. Is that what you mean?

We have not followed or tried any alignment with external event structures. TBH even the [kubernetes-event-exporter](https://github.com/resmoio/kubernetes-event-exporter) does not seem a standard right? Currently no plans to update the events metricset as all work is focused on the otel related work

**Comment by ltflb-bgdi** [comment]
> wed or tried any alignment with external event structures. TBH eve

@gizas, Thanks for answering. So as expected, events are metrics today since metricbeat allowed for easy polling.
Still, as you agree, events should to to logs instead of metrics, at least in OTEL.

Regarding my second question: What I wanted to know is why there is no ecs parsing applied, i.e. only have fields in kubernetes.events which can not be mapped to ecs, as done in most of the log streams.

Since you mentioned as well that k8s events should to to logs instead of metrics, I would highly appreciate if we come up with a logs data_stream for k8s events. As already mentioned, the latest commit conains a kubernetes.event_logs data_stream to ingest logs collected with edot (elastic-otel-collector) k8sobjects receiver, running as a k8s deployment.

The newly created data_stream addresses the following issues.
- Missing k8s cluster-name in logs makes log analysis very hard in environments with multiple k8s clusters.
- Avoid confusion. Having event logs in metrics, while all other logs in logs is confusing and cumbersome.
- Missing ecs compliance makes log-analysis and -correlation complex.

Looking forward for your answers.

**Comment by gizas** [comment]
Thanks for the ideas @ltflb-bgdi !

To clarify the kubernetes.events are part of the kubernetes integration that is being installed with the elastic-agent setups
In otel elastic agent distributions, the events are being collected with k8sobject receiver and are already part of logs.

The otel work is in progress and indeed we are making sure that we can safeguard the quality gates also for events.

For the integration now, we dont want at the moment to introduce such breaking changes, especially when the events metricset is being used as it is for so many years. Hope it makes sense.

**Comment by ltflb-bgdi** [comment]
> Thanks for the ideas @ltflb-bgdi !
> 
> To clarify the kubernetes.events are part of the kubernetes integration that is being installed with the elastic-agent setups In otel elastic agent distributions, the events are being collected with k8sobject receiver and are already part of logs.
> 
> The otel work is in progress and indeed we are making sure that we can safeguard the quality gates also for events.
> 
> For the integration now, we dont want at the moment to introduce such breaking changes, especially when the events metricset is being used as it is for so many years. Hope it makes sense.

**Comment by ltflb-bgdi** [comment]
> Thanks for the ideas @ltflb-bgdi !
> 
> To clarify the kubernetes.events are part of the kubernetes integration that is being installed with the elastic-agent setups In otel elastic agent distributions, the events are being collected with k8sobject receiver and are already part of logs.
> 
> The otel work is in progress and indeed we are making sure that we can safeguard the quality gates also for events.
> 
> For the integration now, we dont want at the moment to introduce such breaking changes, especially when the events metricset is being used as it is for so many years. Hope it makes sense.

@gizas, Thanks for the answer.

To clarify the clarification:

As far as I understand, the config available on Kibana (https://kibana-dev.geobax.tech/app/observabilityOnboarding/otel-logs/?category=logs) does not contain a k8sobjects receiver config and the logs pipeline only contains a filelogreceiver input mention k8sobjects receiver at all.
As well, the collector is only running as a deamonSet, while according to the [k8sobjects receiver documentation](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8sobjectsreceiver#deployment) objects (events) should be collected with a collector running as a deployment of one replica to avoid duplicated events.
Since the config on Kibana does not contain these settings, please let me know where I can get the correct one.

**Comment by gizas** [comment]
Indeed you are right, the events should rely on a deployment as can be collected from a single endpoint. 

The events configuration will also be documented officially from elastic side as we progress. 
FYI I will cc some 'unofficial' helm chart examples that include events and we use in our testing. Mind that this can change if/when we find any updates:

- https://github.com/elastic/k8scollector/blob/otel-mode-test/helm/deployment_events.yaml
- https://github.com/elastic/k8scollector/blob/otel-mode-test/helm/general_dashboard/deploy-otel-collector.yaml

**Comment by ltflb-bgdi** [comment]
> Indeed you are right, the events should rely on a deployment as can be collected from a single endpoint.
> 
> The events configuration will also be documented officially from elastic side as we progress. FYI I will cc some 'unofficial' helm chart examples that include events and we use in our testing. Mind that this can change if/when we find any updates:
> 
> * https://github.com/elastic/k8scollector/blob/otel-mode-test/helm/deployment_events.yaml
> * https://github.com/elastic/k8scollector/blob/otel-mode-test/helm/general_dashboard/deploy-otel-collector.yaml

@gizas, Thanks for confirming and for the links. Unfortunately they return 404. I guess I have no access-rights to this repo.

**Pull request closed without merging**